Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM golang:1.17 as builder
+
+ENV GOPROXY=https://goproxy.cn
+
+WORKDIR /devstream
```

Comment:
After second thought, I think there might be a few issues with building in docker:

## 1. Network Environment

Building inside docker (instead of on their machines directly) for users behind a corporate proxy might cause a `certificate signed by unknown authority` issue, depending on their corporate proxy setup. This wouldn't be a big issue if they built it locally because the chances are, they've already solved it locally before (like when they used `curl`, for example).

## 2. Cross Platform

In essence, the build inside the docker is for Debian/Ubuntu (depending on the base image); but after the build, the binary is copied to the host of the user and will be executed there, and the users' host might differ; maybe it's Mac, maybe it's Windows.

Theoretically, we can write some shell script to detect if the user needs proxy, and what's the user's OS, then call different build commands like:

- `make build-in-docker-without-cn-proxy-debian`
- `make build-in-docker-without-cn-proxy-darwin`
- `make build-in-docker-without-cn-proxy-windows`

- `make build-in-docker-with-cn-proxy-debian`
- `make build-in-docker-with-cn-proxy-darwin`
- `make build-in-docker-with-cn-proxy-windows`

This might be a bit redundant, though.

I'm not completely sure if this is the simplest way to build a CLI tool.

@lfbdev

Reply:
Yes, I also have the idea to support cross-platform compile, but support the linux/amd64 first, and I'll add mac support recently